### PR TITLE
Vormetric Keys for Storage / Database

### PR DESCRIPTION
**Pull request by ronaldheft-usds**
This PR rotates our encryption to use the RSA keys provided by Vormetric.

## Changes
- Rotates storage account to use the Vormetric key
- Rotates database to use the Vormetric key

## Checklist
- [ ] Tested locally?
- [ ] Added new dependencies?
- [ ] Updated the docs?
- [ ] Added a test?
- [ ] Did you check for sensitive data, and remove any?
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

## Fixes 

- #issue

## To Be Done

- Once this is verified in test, we reach out to the IAMB team, they send us keys in prod, we update our config to use in prod, then deploy.

### CODE REVIEW

**Comment by ronaldheft-usds** [comment]
@cglodosky I could use your assistance testing this in our test environment. We can investigate after #521.

File: operations/app/src/modules/database/main.tf [hunk 1]
```diff
@@ -27,6 +27,18 @@ resource "azurerm_postgresql_server" "postgres_server" {
   }
 }
 
+data "azurerm_key_vault_key" "postgres_server_encryption_key" {
+  count = var.environment == "test" ? 1 : 0 // Only use Vormetric keys in test
```
Comment (cglodosky) [inline operations/app/src/modules/database/main.tf:31]: Is this just until we are fully ready to roll out to production?  Or did I miss something that we aren't doing this in prod?

File: operations/app/src/modules/storage/main.tf [hunk 2]
```diff
@@ -23,6 +23,14 @@ resource "azurerm_storage_account" "storage_account" {
   }
 }
 
+resource "azurerm_storage_account_customer_managed_key" "storage_key" {
+  count = var.environment == "test" ? 1 : 0 // Only use Vormetric keys in test
+  key_name = "${var.resource_prefix}-key"
```
Comment (cglodosky) [inline operations/app/src/modules/storage/main.tf:28]: For continuity, do we want to put the number of bits (ie. 2048, 4096) into the key name?  Or is everything 4096 except for the one above?

**Review by cglodosky — COMMENTED** [review]
I left two questions as comments, but the rest looks good to me.

File: operations/app/src/modules/database/main.tf (hunk 1 above)
Comment (ronaldheft-usds) [reply operations/app/src/modules/database/main.tf:31]: Yes, it's only in test for the moment. We need to have the Vormetric team push keys to prod once we validate the environment.

File: operations/app/src/modules/storage/main.tf (hunk 2 above)
Comment (ronaldheft-usds) [reply operations/app/src/modules/storage/main.tf:28]: I'll ask if they can rename the key. Also going to ask about adding it to staging once we get the staging environment up.

File: operations/app/src/modules/database/main.tf (hunk 1 above)
Comment (ronaldheft-usds) [reply operations/app/src/modules/database/main.tf]: I updated this to more easily change which environments are used by setting a variable that passes through.

File: operations/app/src/modules/storage/main.tf (hunk 2 above)
Comment (ronaldheft-usds) [reply operations/app/src/modules/storage/main.tf]: Going to work on getting the key changed in test, but the key names are clearer in staging, and will be in prod.

**Comment by ronaldheft-usds** [comment]
Aiming to test around 3:30pm EST, after @jimduff-usds gives me an overview of running smoke tests in staging.

**Review by cglodosky — APPROVED** [review]
Looks good to me.

**Pull request merged**